### ISSUE TITLE
load_weights by_name didn't work

### ISSUE THREAD

**Issue by dddzg** [keras-team/keras#9683]
there is a key part of my code.(the weights file is downloaded from url in the source code)
```
from keras.applications import Xception
cnn_model = Xception(include_top=False, input_shape=(width, width, 3), weights=None)
cnn_model.load_weights('./xception_weights_tf_dim_ordering_tf_kernels_notop.h5')
```
it works **alright**, and loss is good.

However,when I only modify the last line of code to
`cnn_model.load_weights('./xception_weights_tf_dim_ordering_tf_kernels_notop.h5',by_name=True)`

there is no error or any warning,however, ths loss is much **worse** than before(just like random initial).

I suspect whether the parameter **by_name=True** works right.is this a bug or just my fault?

(I am going to modify the Xception structure and use pre-trained weights.So,i try this code.

**Comment by dddzg** [comment]
and is there any elegent way to use pre-trained weights when I want to change some layers in the Xception?

**Comment by dddzg** [comment]
finally,I find this problems by read the 'xception_weights_tf_dim_ordering_tf_kernels_notop.h5' file.
The layer name in the .h5 file is 'convolution2d_1','convolution2d_2','convolution2d_3'
but the layer name in the keras code is 'block1_conv1','block1_conv2'.
so it can't load the weight properly.

I think it is a mistake,maybe we can update the .h5 file? or give some warning?

**Comment by luvWY** [comment]
I wonder to know why.. because when you code 'cnn_model.summary()', the layer name is fixed.. and the .h5 file is also fixed. But I used 'load_weights('./xception_weights_tf_dim_ordering_tf_kernels_notop.h5')', the weights is corrected, and used 'load_weights('./xception_weights_tf_dim_ordering_tf_kernels_notop.h5',by_name=True)', the weights is incorrected. Why.. how to find load_weights source code. did it change the layer name or name in .h5 file??

**Comment by harshitadd** [comment]
Has this been resolved yet? 
I am trying to load weights into a new model having a different input layer ( named differently as the one in the original model ) but by_name and skip_match do not seem to be working as I am getting a shape mismatch error for the weights of the new input layer despite it having a different name.
